src/Mutator/Operator/SpreadAssignment.php

```diff
@@ -0,0 +1,107 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Mutator\Operator;
+
+use function count;
+use Infection\Mutator\Definition;
+use Infection\Mutator\GetMutatorName;
+use Infection\Mutator\Mutator;
+use Infection\Mutator\MutatorCategory;
+use PhpParser\Node;
+use Webmozart\Assert\Assert;
+
+/**
+ * @internal
+ *
+ * @implements Mutator<Node\Expr\Array_>
+ */
+final class SpreadAssignment implements Mutator
+{
+    use GetMutatorName;
+
+    public static function getDefinition(): ?Definition
+    {
+        return new Definition(
+            <<<'TXT'
+Removes a spread operator in an array expression and turns it into an assignment. For example:
+
+```php
+$x = [...$collection];
+```
+
+Will be mutated to:
+
+```php
+$x = $collection;
+```
+TXT
+            ,
+            MutatorCategory::SEMANTIC_REDUCTION,
+            null,
+            <<<'DIFF'
+- $x = [...$collection];
++ $x = $collection;
+DIFF
+        );
+    }
+
+    /**
+     * @psalm-mutation-free
+     *
+     * @return iterable<Node\Expr>
+     */
+    public function mutate(Node $node): iterable
+    {
+        Assert::allNotNull($node->items);
+
+        yield $node->items[0]->value;
+    }
+
+    public function canMutate(Node $node): bool
+    {
+        if (!$node instanceof Node\Expr\Array_) {
+            return false;
+        }
+
+        if (count($node->items) !== 1) {
+            return false;
+        }
+
+        Assert::allNotNull($node->items);
+
+        return $node->items[0]->unpack;
```

Comment:
the assertion is more for PHPStan because from what I understand and see in other examples the items will be `ArrayItem`, but the type [here](https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/Node/Expr/Array_.php#L13) is `ArrayItem|null`. Let me know if I'm missing something, it's the first time I'm working with the php parser